File: Cryptographic_Frontend/cf/GetAttributeMethod.cpp
/**
 * @author Francisco Cifuentes [francisco@niclabs.cl]
 */

#include "GetAttributeMethod.hpp"
#include "Argument.hpp"
#include "ResponseMessage.hpp"

#include <json/json.h>

cf::GetAttributeMethod::GetAttributeMethod(std::string attribute, std::string handler)
  : Method("GetAttribute")
{
  addArgument(ArgumentPtr(new StringArgument("attribute", attribute)));
  addArgument(ArgumentPtr(new StringArgument("handler", handler)));
}

cf::ResponseMessagePtr cf::GetAttributeMethod::parseResponse(const std::string & message) {
    Json::Value json;
    Json::Reader reader;
    
    reader.parse(message, json);
    ResponseMessagePtr response(new ResponseMessage());

    response->addValue(ArgumentPtr( new StringArgument("attributeValue", json["attributeValue"].asString()) ));

    return response;
}
// kate: indent-mode cstyle; indent-width 4; replace-tabs on;

File: Cryptographic_Backend/src/main/java/cl/niclabs/cb/backend/methods/GetAttributeMethod.java
package cl.niclabs.cb.backend.methods;

import cl.niclabs.cb.backend.Method;

public abstract class GetAttributeMethod implements Method {
	static public class ReturnValue {
		public final String attributeValue;

        public ReturnValue(String attributeValue) {
            this.attributeValue = attributeValue;
        }
    }
	static public class Args {
		public final String attribute;
		public final String handler;

        public Args(String attribute, String handler) {
            this.attribute = attribute;
            this.handler = handler;
        }
    }
	protected final String attribute;
	protected final String handler;
	
	public GetAttributeMethod(Args args) {
		this.attribute = args.attribute;
		this.handler = args.handler;
	}

}
